### DISCUSSION TITLE
Use FQCN in type error message if two or more types have the same name

### DISCUSSION THREAD

**Discussion by hnrklssn** [scala/scala3#17842] [Feature Requests]
I recently encountered a confusing type error message because I had absentmindedly named a class `String` (details [here](https://stackoverflow.com/questions/68146374/why-cant-a-string-literal-be-assigned-to-variable-of-type-string/68146434)). The error message did not clarify however that the two types named were in fact not the same class. As this was the first time trying out Scala 3, I thought it was due to some new dependent typing for string literals, but couldn’t figure out the proper syntax for it. I think it would have been quite helpful if the error message replaced the type names with their FQCNs when they clash with another one. Potentially even if that other class is not named in the message, but is available in the current scope, just to be explicit.

**Comment by bishabosha** [comment]
moved to https://github.com/lampepfl/dotty/issues/18678
